core/src/main/java/io/questdb/std/IntList.java

```diff
@@ -59,6 +60,10 @@ public void addAll(IntList that) {
         System.arraycopy(that.data, 0, this.data, p, s);
     }
 
+    public void allocate(int size) {
```

Comment:
nit: in `LongList` a similar public method is called `checkCapacity`

Reply:
Docs here too please.

Is `size` additional capacity? Is it the new expected total? Will it be respected exactly or can it grow over?


..it also it seems to just call `checkCapacity`.

Personal side rant: (I expect no action) I really dislike the name `checkCapacity`: It reads like it would through it's a validation step that would throw an exception if there isn't enough. It would be good to rename all our `checkCapacity` to `ensureSpareCapacity` or some other better name.

Reply:
> I really dislike the name `checkCapacity`

This name is used in many places across the code base. If we decide to change it, better to do that in a separate PR.